internal/util/fetch/fetch.go

```diff
@@ -62,34 +76,183 @@ func (c Command) validate(kf *kptfilev1.KptFile) error {
 	if kf.Upstream == nil {
 		return errors.E(op, errors.MissingParam, fmt.Errorf("kptfile doesn't contain upstream information"))
 	}
-	switch kf.Upstream.Type {
-	case kptfilev1.GitOrigin:
-		if kf.Upstream.Git == nil {
-			return errors.E(op, errors.MissingParam, fmt.Errorf("kptfile upstream doesn't have git information"))
-		}
 
-		g := kf.Upstream.Git
-		if len(g.Repo) == 0 {
-			return errors.E(op, errors.MissingParam, fmt.Errorf("must specify repo"))
-		}
-		if len(g.Ref) == 0 {
-			return errors.E(op, errors.MissingParam, fmt.Errorf("must specify ref"))
-		}
-		if len(g.Directory) == 0 {
-			return errors.E(op, errors.MissingParam, fmt.Errorf("must specify directory"))
-		}
+	if kf.Upstream.Git == nil {
+		return errors.E(op, errors.MissingParam, fmt.Errorf("kptfile upstream doesn't have git information"))
+	}
 
-	case kptfilev1.OciOrigin:
-		if kf.Upstream.Oci == nil {
-			return errors.E(op, errors.MissingParam, fmt.Errorf("kptfile upstream doesn't have oci information"))
-		}
-		if len(kf.Upstream.Oci.Image) == 0 {
-			return errors.E(op, errors.MissingParam, fmt.Errorf("must specify image"))
+	g := kf.Upstream.Git
+	if len(g.Repo) == 0 {
+		return errors.E(op, errors.MissingParam, fmt.Errorf("must specify repo"))
+	}
+	if len(g.Ref) == 0 {
+		return errors.E(op, errors.MissingParam, fmt.Errorf("must specify ref"))
+	}
+	if len(g.Directory) == 0 {
+		return errors.E(op, errors.MissingParam, fmt.Errorf("must specify directory"))
+	}
+	return nil
+}
+
+// cloneAndCopy fetches the provided repo and copies the content into the
```

Comment:
Just a note - this would be a good file to bookmark - that in a PR to main if there are no diffs or if there are that the main is authoritative.

Unless the text of the code came from the current main (rather than a "commit revert") - then having no diffs should be almost certain.

Reply:
good point. I will still need to be very careful with the merge into main.

Reply:
Confirmed. this file is near identical; only diff relative to main are filesystem changes.